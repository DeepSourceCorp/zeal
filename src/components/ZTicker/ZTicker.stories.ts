import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'
import ZTicker from './ZTicker.vue'

export default {
  title: 'Ticker',
  component: ZTicker,
  excludeStories: /.*Data$/
}

export const DefaultTicker = () => ({
  components: { ZTicker },
 
  template: `<div class='padded-container'>
        <div class="input-container">
            <z-ticker></z-ticker>
        </div>
    </div>`
})

