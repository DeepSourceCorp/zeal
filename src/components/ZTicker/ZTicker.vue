<template>
  <div class="z-indicator flex items-center">
    <div class="flex items-center space-x-0.5"
        :class="[`text-${color}`]">
      <span>{{value}}%</span>
      <z-icon icon="triangle-down" :color="color" size="small" class="transform"
              :class="{
                'rotate-180 mt-0.5': this.type === 'increase',
                'rotate-0': this.type === 'decrease',
                'hidden': this.hideIcon
              }"></z-icon>
    </div>
    <slot name="content"></slot>
  </div>
</template>

<script>
import ZIcon from '@/components/ZIcon/ZIcon.vue'

export default {
  name: 'ZTicker',
  components: {
    ZIcon 
  },
  props: {
    value: {
        type: [String, Number],
        default: 100
    },
    type: {
        type: String,
        default: 'increase'
    },
    hideIcon: {
      type: Boolean
    }
  },
  computed: {
    color() {
      const colors = {
        increase: 'juniper',
        decrease: 'cherry'
      }
      return colors[this.type]
    }
  }
}
</script>
