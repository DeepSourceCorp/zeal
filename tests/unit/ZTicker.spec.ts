import ZTicker from '../../src/components/ZTicker'

import { shallowMount } from '@vue/test-utils'

describe('Ticker', () => {
  it('renders a default ticker', () => {
    const wrapper = shallowMount(ZTicker)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a negative type ticker', () => {
    const wrapper = shallowMount(ZTicker, {
      propsData: {
        type: 'decrease',
        value: '44%'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a ticker with additional content', () => {
    const wrapper = shallowMount(ZTicker, {
      propsData: {
        type: 'decrease',
        value: '44%'
      },
      slots: { content: '<span class="text-sm">since last week</span>' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
