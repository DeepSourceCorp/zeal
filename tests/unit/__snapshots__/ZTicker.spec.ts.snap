// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker renders a default ticker 1`] = `
<div class="z-indicator flex items-center">
  <div class="flex items-center space-x-0.5 text-juniper"><span>100</span>
    <z-icon-stub icon="triangle-down" size="x-small" position="" color="juniper" class="transform rotate-180 mt-px"></z-icon-stub>
  </div>
</div>
`;

exports[`Ticker renders a negative type ticker 1`] = `
<div class="z-indicator flex items-center" type="decrease">
  <div class="flex items-center space-x-0.5 text-juniper"><span>44%</span>
    <z-icon-stub icon="triangle-down" size="x-small" position="" color="juniper" class="transform rotate-180 mt-px"></z-icon-stub>
  </div>
</div>
`;

exports[`Ticker renders a ticker with additional content 1`] = `
<div class="z-indicator flex items-center" type="decrease">
  <div class="flex items-center space-x-0.5 text-juniper"><span>44%</span>
    <z-icon-stub icon="triangle-down" size="x-small" position="" color="juniper" class="transform rotate-180 mt-px"></z-icon-stub>
  </div> <span class="text-sm">since last week</span>
</div>
`;
